### DISCUSSION TITLE
Good way to add AuthUser globally but allow each page to customize options

### DISCUSSION THREAD

**Discussion by pesty** [gladly-team/next-firebase-auth#603] [Q&A]
Hi folks,

I have an app that I have successfully used next-firebase-auth to auth. However, when I try to optimize the experience further, I am running into some challenges and want to ask for advices:

- In order to simplify for most pages, I add `withAuthUser` to _app.js. This way, most people could use `AuthUser` from the context.
- For some pages, I want to optimize the options like `whenUnauthedBeforeInit` so that the user can see the loader (or no loader). One simple way to do that is simply add `withAuthUser` to that page.
- However, I notice that when I do that, there are two login requests being sent. I suspect one is tied to _app and the other one is to the actual page. Preferably I don't want it to send unnecessary login request.

My question:
- Is there a way to suppress the additional login request when I have `withAuthUser` on both `_app.js` and the actual page?
- If not, is there a good pattern to allow the child page to pass or override the options passed to `withAuthUser` in `_app.js`?

Thank you!

**Comment by kmjennison** [comment]
Good question. No, this isn't currently supported by the library, unfortunately, though I welcome suggestions on supporting it.
